### PR TITLE
Add in a new fetching of roles in the group sharing tab

### PR DESCRIPTION
**Pull request by lamtu1**
`FEATURE:` Add in a new set of roles into the `share tab` and under the `share group` when create/edit a work.

### CODE REVIEW

**Comment by lamtu1** [comment]
fixes #3044

**Comment by lamtu1** [comment]
`IMAGE:` Shown to populate in the sharing group field

Screenshot 2024-02-14 at 9 20 22 AM

File: app/helpers/hyrax/hyrax_helper_behavior.rb [hunk 1]
```diff
@@ -0,0 +1,383 @@
+# frozen_string_literal: true
+
+# OVERRIDE: Override the file from Hyrax to modify methods usage
```
Comment (CGillen) [inline app/helpers/hyrax/hyrax_helper_behavior.rb:3]: I can't tell what part you modified from this comment. In the future please put the `OVERRIDE` comment near the code that changed.

File: app/helpers/hyrax/hyrax_helper_behavior.rb [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/AbcSize
+# rubocop:disable Metrics/MethodLength
+module Hyrax
+  # Hyrax Helper Behavior
+  module HyraxHelperBehavior
```
Comment (CGillen) [inline app/helpers/hyrax/hyrax_helper_behavior.rb:9]: We already have access to the file that includes this behavior:
`app/helpers/hyrax_helper.rb`
and more preferably: 
`app/helpers/oregon_digital/hyrax_helper_behavior.rb` <= please make your edits here

**Review by CGillen — CHANGES_REQUESTED** [review]
We

File: app/helpers/hyrax/hyrax_helper_behavior.rb (hunk 1 above)
Comment (lamtu1) [reply app/helpers/hyrax/hyrax_helper_behavior.rb]: I will keep that in mind for next PR to show where the changes happen.

**Review by CGillen — APPROVED** [review]

**Pull request merged**